projects/coordinator/src/main/java/org/batfish/coordinator/WorkMgr.java

```diff
@@ -811,10 +814,38 @@ public void initTestrig(Path testrigDir,  Path srcDir, boolean autoProcess) {
       CommonUtil.copy(subFile, target);
     }
 
-    if (autoProcess) {
-      //TODO: any logic to automatically parse, analyze the testrig should go here
-      //This can work by creating proper workitems, as done by Client, and calling queueWork
-      throw new BatfishException("Auto processing is not currently implemented");
+    if (autoAnalyze) {
+      List<WorkItem> autoWorkQueue = new LinkedList<>();
+
+      WorkItem parseWork = WorkItemBuilder.getWorkItemParse(containerName, testrigName, false);
+      autoWorkQueue.add(parseWork);
+
+      Set<String> analysisNames = listAnalyses(containerName);
+      for (String analysis : analysisNames) {
+        WorkItem analyzeWork =
+            WorkItemBuilder.getWorkItemRunAnalysis(
+                analysis,
+                containerName,
+                testrigName,
+                BfConsts.RELPATH_DEFAULT_ENVIRONMENT_NAME,
+                null,
+                null,
+                false,
+                false);
+        autoWorkQueue.add(analyzeWork);
+      }
+
+      // NB: This way of doing things only works when we have a single worker; otherwise workitems
+      // lower down the order can get fired before those higher in the order
+      // The right solution is to put workitem2 on the queue only after workitem1 has finished
+      // successfully. The rightest solution is for workers to be aware of dependencies so they
+      // don't try to execute tasks that depend on other tasks that are currently being executed.
+
+      for (WorkItem workItem : autoWorkQueue) {
```

Comment:
Is there a way to bail here if there's more than 1 worker connected?

Reply:
Being tracked as #573